<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
<HEAD>
 <TITLE>30046</TITLE>
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <script type="text/javascript">
  <!--
   function ZweiFrames(URL1,F1,URL2,F2)
   {
    parent.frames[F1].location.href=URL1;
    parent.frames[F2].location.href=URL2;
   }
  //-->
  </script>
</HEAD>
<BODY BGCOLOR="#ffffff" style="margin-left:25">
<HR>
<H3><CENTER>Week8_Task2.java</CENTER></H3><HR>
<PRE>
<A NAME="0"></A>import java.io.File;
import java.io.FileNotFoundException;
<FONT color="#0000ff"><div style="position:absolute;left:0"><A HREF="javascript:ZweiFrames('match60897-0.html#0',2,'match60897-top.html#0',1)"><IMG SRC="back.gif" ALT="other" BORDER="0" ALIGN="left"></A></div><B>import java.io.FileReader;
import java.io.IOException;

public class Week8_Task2 {
    void NPEx() throws NullPointerException {
        throw new NullPointerException(&quot;Lỗi Null Pointer&quot;);

    }

    String NPExTest() {
        try {
            NPEx();
        } catch (NullPointerException e) {
            return e.getMessage();
        }
        return &quot;Không có lỗi&quot;;
    }

    void AIOoBEx() throws ArrayIndexOutOfBoundsException {
        throw new ArrayIndexOutOfBoundsException(&quot;Lỗi Array Index Out of Bounds&quot;);

    }

    String AIOoBExTest() {
        try {
            AIOoBEx();
        } catch (ArrayIndexOutOfBoundsException e) {
            return e.getMessage();
        }
        return &quot;Không có lỗi&quot;;
    }

    void ArithEx() throws ArithmeticException {
        throw new ArithmeticException(&quot;Lỗi Arithmetic&quot;);

    }

    String ArithExTest() {
        try {
            ArithEx();
        } catch (ArithmeticException e) {
            return e.getMessage();
        }
        return &quot;Không có lỗi&quot;;
    }

    void FNFEx() throws FileNotFoundException {
        throw new FileNotFoundException(&quot;Lỗi File Not Found&quot;);

    }

    String FNFExTest() {
        try {
            FNFEx();
        } catch (FileNotFoundException e) {
            return e.getMessage();
        }
        return &quot;Không có lỗi&quot;;
    }

    void IOEx() throws IOException {
        throw new IOException(&quot;Lỗi IO&quot;);
    }

    String IOExTest() {
        try {
            IOEx();
        } catch (IOException e) {
            return e.getMessage();
        }
        return &quot;Không có lỗi&quot;;
    }


}</B></FONT>
</PRE>

</BODY>
</HTML>
